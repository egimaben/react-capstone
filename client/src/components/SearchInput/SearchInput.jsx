import React from "react";
import PropTypes from "prop-types";
import "./SearchInput.css";
const SearchInput = ({ onSearchQueryChange, label, value }) => {
  return (
    <div className="animated-search-input">
      <label>
        <input
          type="text"
          value={value}
          required
          onChange={onSearchQueryChange}
        />
        <div className="label-text">
          <i className="fa fa-search" />
          {label}
        </div>
      </label>
    </div>
  );
};
SearchInput.propTypes = {
  onSearchQueryChange: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired
};
export default SearchInput;
