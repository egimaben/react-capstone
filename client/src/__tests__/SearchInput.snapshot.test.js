import React from 'react';
import renderer from 'react-test-renderer'
import SearchInput from '../components/SearchInput';

console.error = err => { throw new Error(err); };
console.warn = warning => { throw new Error(warning); };

it('card renders correctly', () => {
    
  const rendered = renderer.create(<SearchInput onSearchQueryChange={jest.fn()} label='search' value='value' />);
  expect(rendered.toJSON()).toMatchSnapshot();
});