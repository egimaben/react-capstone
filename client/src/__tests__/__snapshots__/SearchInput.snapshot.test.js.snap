// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card renders correctly 1`] = `
<div
  className="animated-search-input"
>
  <label>
    <input
      onChange={[MockFunction]}
      required={true}
      type="text"
      value="value"
    />
    <div
      className="label-text"
    >
      <i
        className="fa fa-search"
      />
      search
    </div>
  </label>
</div>
`;
